src/sagemaker/serve/builder/model_builder.py

```diff
@@ -672,3 +682,58 @@ def validate(self, model_dir: str) -> Type[bool]:
         """
 
         return get_metadata(model_dir)
+
+    def total_inference_model_size_mib(self):
+        """Calculates the model size from HF accelerate
+
+        This function gets the model size from accelerate. It also adds a
+        padding and converts to size MiB. When performing inference, expect
+        to add up to an additional 20% to the given model size as found by EleutherAI.
+        """
+        dtypes = "float32"
+        try:
+            if self.env_vars.get("dtypes"):
+                dtypes = self.env_vars.get("dtypes")
+
+            parser = estimate_command_parser()
+            args = parser.parse_args([self.model, "--dtypes", dtypes])
+        except ValueError:
+            logging.error("Args specified incorrect for model %s", self.model)
+
+        output = gather_data(
+            args
+        )  # "dtype", "Largest Layer", "Total Size Bytes", "Training using Adam"
+
+        total_memory_size_mib = MEMORY_BUFFER_MULTIPLIER * output[0][2] * MIB_CONVERSION_FACTOR
+        logger.info("Total memory size MIB: %s", total_memory_size_mib)
+        return total_memory_size_mib
+
+    def can_fit_on_single_gpu(self):
+        """Check if model can fit on a single GPU
+
+        This function gets the GPU info or fallback to set the size of a single GPU.
+        If the size of the model is <= single gpu memory size, returns true.
+        """
+        try:
+            gpu_info = _get_gpu_info(self.instance_type, self.sagemaker_session)
+            logger.info("GPU info %s for instance %s", gpu_info, self.instance_type)
+            single_gpu_size_mib = gpu_info[1] / gpu_info[0]
+        except ValueError:
+            gpu_fallback = _get_gpu_info_fallback(
+                self.instance_type, self.sagemaker_session.boto_region_name
+            )
+            logger.info("GPU fallback picked up %s", gpu_fallback)
+            single_gpu_size_mib = gpu_fallback[1] / gpu_fallback[0]
+
+        if single_gpu_size_mib is None:
```

Comment:
This won't be `None` right? Since `_get_gpu_info` and `_get_gpu_info_fallback` either return some values or throw excpetions.

Reply:
Both will throw `ValueError` if gpu info is not found for the given instance. Please handle the exception for the fallback as well.

Reply:
Yes that's correct, addressed